### DISCUSSION TITLE
Make VIRTUAL_PREFIX configurable

### DISCUSSION THREAD

**Discussion by marpe** [posva/unplugin-vue-router#452] [Questions and Help]
I'm using this in the development of a chrome web extension and am receiving this error:

```
Cannot load extension with file or directory name __vue-router. Filenames starting with "_" are reserved for use by the system.
Could not load manifest.
```

So it would be nice to be able to configure the `VIRTUAL_PREFIX` or maybe opt for another default?

**Comment by posva** [comment]
That looks like a weird limitation. In the end, there is no file named that way It's just a virtual file that is handled by the dev server. Other libraries like uno.css does this as well

It would be helpful to see how this error is produced

**Reply by marpe** [reply]
Removed my `node_modules` which I had patched, and tried to recreate it, but haven't been able to. First time using vue/vite and using a template which already have a bunch configured, so very likely I had something misconfigured at one point (I definitely saw a folder named `__something` in the build output :P).

You can close this and I'll make sure to include a reproducible example if I run into it again.

**Comment by jods4** [comment]
@posva please note that changing `VIRTUAL_PREFIX` is a breaking change, that should be announced when released.
I think the "internal" module name needs to be documented and considered stable.

Example of places where we depend on the name `__vue-router/auto-routes`:
- Whitelisting the module in other Vite plugins config. Specifically our routes meta contain unocss icon names, so we whitelisted `__vue-router/auto-routes` in UnoCSS `include` config.
- Handling HMR updates of the route table: `import.meta.hot.accept("__vue-router/auto-routes", ...)`

**Reply by posva** [reply]
I'm more worried that you have to depend on the internal virtual representation of the module that is only needed in order to write the plugin for vite 😓, I wouldn't even use such representation if it wasn't required by the bundlers implementations. I see that as a bug here, or a design issue in vite, or as a bug in unocss, but I don't know where. Which also means that it will probably change again to fix issues. Although you could use `.endsWith('vue-router/auto-routes')` thought, that should cover any prefixes.

The reason behind the change was needed to support HMR. I worked on this with the help of antfu and followed some of the design choices made in unocss. The documentation on HMR on vite wasn't really effective neither was [this article](https://bjornlu.com/blog/hot-module-replacement-is-easy). I think this is specific to the virtual module though...

**Reply by jods4** [reply]
Yeah, we actually used the glob `**/auto-routes` in UnoCSS include config, precisely because I felt uncomfortable with this undocumented prefix.

I agree 100% that the design feels off.
Not sure what the better way would be, but the intuitive thing is to `include: "vue-router/auto-routes"`, which is how the module is named in JS/TS code. That does not work, but ideally I think it should.

**Comment by posva** [comment]
The prefix is now the official recommendation of vite in c85dd85

Let me know if this is still an issue (#608 )
